packages/select/docs/FilterMultiSelect.stories.tsx

```diff
@@ -334,3 +335,92 @@ export const DefaultKaizenSiteDemoWithoutScrollbar = () => {
 }
 
 DefaultKaizenSiteDemoWithoutScrollbar.storyName = "With no scrollbar"
+
+export const Async: ComponentStory<typeof FilterMultiSelect> = args => {
+  const [open, setOpen] = useState(false)
+  const [selectedPeople, setSelectedPeople] = useState<string[]>([])
+  const [searchState, setSearchState] = useState("")
+  const { data, isLoading, fetchNextPage, isFetchingNextPage, hasNextPage } =
+    useInfiniteQuery(
+      ["startrek-sg1", searchState],
+      ({ pageParam = 1 }) =>
+        fetch(
+          `https://swapi.dev/api/people/?page=${pageParam}&search=${searchState}`
+        ).then(res => res.json()) as Promise<{
+          results: Array<{ name: string; url: string }>
+          next: string
+        }>,
+      {
+        enabled: open,
+        keepPreviousData: true,
```

Comment:
Keeping previous data is important here. Currently the `loadingSkeleton` prop blocks the entire component from being interacted with, including the search input. However, if you try and render something more controlled:
```
            <FilterMultiSelect.SearchInput />
            {isLoading ? ( <Skeleton />) : (
            <>
              <FilterMultiSelect.ListBox>
                {({ allItems }) =>
                  allItems.map(item => (
                    <FilterMultiSelect.Option key={item.key} item={item} />
                  ))
                }
              </FilterMultiSelect.ListBox>
              {hasNextPage && (
                <FilterMultiSelect.LoadMoreButton onClick={() => fetchNextPage()}>
                  {isFetchingNextPage ? "Fetching..." : "Load more"}
                </FilterMultiSelect.LoadMoreButton>
              )}
              <FilterMultiSelect.MenuFooter>
                <FilterMultiSelect.SelectAllButton />
                <FilterMultiSelect.ClearButton />
              </FilterMultiSelect.MenuFooter>
              <>
              )}
            </>
```
The whole component blows up.

Reply:
Also `FilterMultiSelect.MenuLoadingSkeleton` isn't a component. I do not know why.

Reply:
> Also `FilterMultiSelect.MenuLoadingSkeleton` isn't a component. I do not know why.

Hm yeah, that might have been a mistake.